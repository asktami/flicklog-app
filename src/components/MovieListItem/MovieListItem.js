import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import WatchListButton from '../../components/WatchListButton/WatchListButton';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { convertDate } from '../Utils/Utils';

import MISSING_IMG from '../../img/missing_image.png';

// alternative missing image placeholder
// const DEFAULT_PLACEHOLDER_IMAGE = 'https://via.placeholder.com/154';

const MovieListItem = (props) => {
	const { movie, showWatchListButton } = props;

	// update stored movie.background
	// let background_test = DEFAULT_PLACEHOLDER_IMAGE;
	movie.background = movie.poster_path
		? 'https://image.tmdb.org/t/p/w300' + movie.poster_path
		: movie.backdrop_path
		? `https://image.tmdb.org/t/p/w154${movie.backdrop_path}`
		: MISSING_IMG;

	// update stored movie.title
	if (!movie.title) {
		movie.title = movie.original_title;
	}

	// update stored movie.converted_release_date
	// handle that postgres adds a timestamp to the field which I don't want
	if (movie.release_date !== undefined && movie.release_date.includes('T')) {
		movie.converted_release_date = convertDate(
			movie.release_date.split('T')[0]
		);
	} else {
		movie.converted_release_date = convertDate(movie.release_date);
	}

	return (
		<figure>
			<img src={movie.background} alt={movie.title} className="figure-image" />
			<figcaption>
				<Link to={`/movies/${movie.id}`}>{movie.title}</Link>
				<br />
				{movie.converted_release_date}
				{showWatchListButton ? (
					<>
						<br />
						<WatchListButton {...movie} />{' '}
					</>
				) : null}

				<div className="overlay">
					<p className="overlay-text">
						<FontAwesomeIcon icon={['fas', 'star']} size="1x" />
						&nbsp;
						{movie.vote_average * 10}%{' '}
						<span className="votes">{movie.vote_count}&nbsp;votes</span>
					</p>
				</div>
			</figcaption>
		</figure>
	);
};

export default MovieListItem;
