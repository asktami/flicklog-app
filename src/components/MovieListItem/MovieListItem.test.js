import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import MovieListItem from './MovieListItem';
import MISSING_IMG from '../../img/missing_image.png';

describe(`MovieListItem component`, () => {
	const props = {
		movie: {
			poster_path: 'vfrQk5IPloGg1v9Rzbh2Eg3VGyM.jpg',
			backdrop_path: 'AmR3JG1VQVxU8TfAvljUhfSFUOx.jpg',
			title: 'Alien',
			release_date: '1979-05-25',
			vote_average: 8.1,
			vote_count: 5694,
			id: 348,
		},
	};

	it('renders a MovieListItem by default', () => {
		const wrapper = shallow(<MovieListItem {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('renders the MovieListItem given props', () => {
		const wrapper = shallow(<MovieListItem {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
