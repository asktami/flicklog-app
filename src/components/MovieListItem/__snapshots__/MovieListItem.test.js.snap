// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieListItem component renders a MovieListItem by default 1`] = `
<figure>
  <img
    alt="Alien"
    className="figure-image"
    src="https://image.tmdb.org/t/p/w300vfrQk5IPloGg1v9Rzbh2Eg3VGyM.jpg"
  />
  <figcaption>
    <Link
      to="/movies/348"
    >
      Alien
    </Link>
    <br />
    May 25, 1979
    <div
      className="overlay"
    >
      <p
        className="overlay-text"
      >
        <FontAwesomeIcon
          icon={
            Array [
              "fas",
              "star",
            ]
          }
          size="1x"
        />
         
        81
        %
         
        <span
          className="votes"
        >
          5694
           votes
        </span>
      </p>
    </div>
  </figcaption>
</figure>
`;

exports[`MovieListItem component renders the MovieListItem given props 1`] = `
<figure>
  <img
    alt="Alien"
    className="figure-image"
    src="https://image.tmdb.org/t/p/w300vfrQk5IPloGg1v9Rzbh2Eg3VGyM.jpg"
  />
  <figcaption>
    <Link
      to="/movies/348"
    >
      Alien
    </Link>
    <br />
    May 25, 1979
    <div
      className="overlay"
    >
      <p
        className="overlay-text"
      >
        <FontAwesomeIcon
          icon={
            Array [
              "fas",
              "star",
            ]
          }
          size="1x"
        />
         
        81
        %
         
        <span
          className="votes"
        >
          5694
           votes
        </span>
      </p>
    </div>
  </figcaption>
</figure>
`;
